File: JavaScript/src/MajorityElement.js
/**
 * Given an array of size n, find the majority element. The majority element is the element that appears more than ⌊ n/2 ⌋ times.
 *
 * You may assume that the array is non-empty and the majority element always exist in the array.
 *
 * Accepted.
 */

/**
 * @param {number[]} nums
 * @return {number}
 */
let majorityElement = function (nums) {
    let map = [];
    for (let i = 0, len = nums.length; i < len; i++) {
        if (!map[nums[i]]) {
            map[nums[i]] = 1;
        } else {
            map[nums[i]]++;
        }

        if (map[nums[i]] > len / 2) {
            return nums[i];
        }
    }
};

if (majorityElement([2, 1, 1]) === 1) {
    console.log("pass")
} else {
    console.error("failed")
}

if (majorityElement([0, 1, 1, 1, 2, 3, 1]) === 1) {
    console.log("pass")
} else {
    console.error("failed")
}

if (majorityElement([1, 1]) === 1) {
    console.log("pass")
} else {
    console.error("failed")
}

File: Kotlin/src/MajorityElement.kt
/**
 * Given an array of size n, find the majority element. The majority element is the element that appears more than ⌊ n/2 ⌋ times.
 *
 * You may assume that the array is non-empty and the majority element always exist in the array.
 *
 * Accepted.
 */
class MajorityElement {

    fun majorityElement(nums: IntArray): Int {
        val map = mutableMapOf<Int, Int>()

        for (i in nums) {
            if (map.containsKey(i)) {
                map[i] = map[i]!! + 1
            } else {
                map[i] = 1
            }

            if (map[i]!! > nums.size / 2) {
                return i
            }
        }

        return nums[0]
    }

}
